### ISSUE TITLE
Feature request: remember "Try Elm" code across visits

### ISSUE THREAD

**Issue by jwhitfieldseed** [elm-lang/elm-lang.org#434]
It would be great if http://elm-lang.org/try remembered the last-edited program across visits. This is something that the [Babel.js live demo](https://babeljs.io/repl/#?experimental=false&evaluate=true&loose=false&spec=false&code=console.log(%22I%20really%20like%20this%20feature!%22)%3B) does really well.

Babel stores the current program in localStorage and in the URL itself. The latter is especially useful for sharing example programs.

If this is desirable I'll work up a PR for either/both approaches.

**Comment by wavebeem** [comment]
:+1:

**Comment by evancz** [comment]
I don't want to get elm-lang.org into the IDE business. If you want to make a site that stores this kind of stuff, go for it! The website is just to help you get started and transitioned to doing it on your actual computer.

Someone made http://www.share-elm.com a while ago to make sharing code nice, but they have not been keeping up with Elm releases, so there is a niche to fill here!

**Comment by jwhitfieldseed** [comment]
I don't think this brings Try Elm into IDE territory any more than, say, the type hints/doc links (which are so useful, btw). It's about continuity for people who enjoy the zero impedance of sketching code in the browser. In case I didn't make this clear, I'm talking client-side storage only.

FWIW, this is my motivation:

* The Try Elm page is __excellent__. It's one of the things that got me excited about Elm in the first place. It shows that this is a powerful, accessible language that is built for the front end. 

 The only thing I don't like is that I can't work an idea over several sessions. I'm new to Elm and FP, so sometimes it takes me some thinking time to grasp a new concept. It would be useful to return to where I left off after a Eureka moment.

* Playing with an example in the browser is a natural way to get started with Elm, particularly for JS devs. 

 I did that with the [binary tree](http://elm-lang.org/examples/binary-tree) challenges. I got into it and refined my functions over a few sessions. I pasted my code in each time, and I lost progress a couple of times when I accidentally closed a tab and my browser crashed (infinite recursion may been involved :boom:). 

  Recalling the last edit would have helped me out there. Regardless I found the online experience powerful and fun enough that I got pretty well into the language without having to install anything at all. To me, as an Elm beginner, this is something that should be encouraged. It didn't deter me from downloading the toolchain; it made me _excited_ to do it. 

* I agree that the website isn't a substitute for downloading. I've done that too, of course, but I prefer the browser for building an intuition for the language and sketching ideas. In this case I don't care about saving my code forever, just long enough to _understand_ it. 

* I often use [links to the Babel live demo](https://github.com/jwhitfieldseed/es6-talk) to share ES6 snippets. It's a useful tool for teaching people, and for teams working with the language.

Having said that, I get your argument. If you keep this closed I'll hack something with a userscript for now. I would like to build something but I can't commit the time right now.

Thanks! I'm looking forward to exploring Elm some more.
